core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -367,6 +360,20 @@ private PendingStream(PickSubchannelArgs args, ClientStreamTracer[] tracers) {
       this.tracers = tracers;
     }
 
+    @Override
+    public void start(ClientStreamListener listener) {
```

Comment:
I need to look into why this change was made. It looks like it delays registering the stream in pendingStreams to avoid createRealStream() from being called. We may need to double-check the transport state here, because there are state checks before `createPendingStream()` that are now out-of-date.